OpenRA.Game/Game.cs

```diff
@@ -189,8 +192,20 @@ public static void RestartGame()
 			// Restart the game with the same replay/mission
 			if (replay != null)
 				JoinReplay(replayName);
-			else
+			else if (mission)
 				StartMission(uid, globalSettings.GameSpeedType, globalSettings.Difficulty);
+			else
+			{
+				JoinServer(IPAddress.Loopback.ToString(), CreateLocalServer(map), "");
+				LobbyInfoChanged += WidgetUtils.Once(() =>
```

Comment:
I suggest splitting this into a method that takes the game speed and difficulty and returns a LobbyInfo, and then a common method that takes a lobbyinfo and sets up and starts a game.